python/src/indexing.cpp

```diff
@@ -400,6 +400,12 @@ array mlx_get_item_nd(array src, const nb::tuple& entries) {
     src = slice(src, starts, ends, strides);
   }
 
+  bool none_exists = false;
+  for (auto idx : remaining_indices) {
+    if (idx.is_none())
+      none_exists = true;
+  }
```

Comment:
Instead of having this as a separate loop, you can integrate it into the previous loop to check for Nones